File: solutions/javascript/grains.js
/*
  Desafio do Exercism: https://exercism.org
  Descrição do problema disponível:
  https://exercism.org/tracks/javascript/exercises/grains
  Data: feito a 5 meses atrás
*/

export const square = (sq) => {
  if(sq <= 0 || sq > 64) throw new Error('square must be between 1 and 64')
  return 2n ** BigInt(sq - 1)
}
export const total = () => (18446744073709551615n)

File: solutions/python/grains.py
'''
  Desafio do Exercism: https://exercism.org
  Descrição do problema disponível:
  https://exercism.org/tracks/python/exercises/grains
  Data: 06/2024
'''

def square(number):
    if number > 64 or number < 1:
        raise ValueError("square must be between 1 and 64")
    return 2 ** (number - 1)
    
def total():
    return (2 ** 64) - 1